### PR TITLE
Restructure Documentation

### PR DESCRIPTION
**Pull request by smithis7**
This is part 1 of a series of PRs restructuring the documentation. Please view the built documentation [here](https://smithis7.github.io/tardis/branch/restructure/index.html).

**Description**

There are a lot of changes made. The overarching themes of these changes are as follows:
- Removes "using TARDIS" section
- Creates "I/O" section which is differently organized
- Splits the configuration documentation into many different pages, and incorporates the atomic and models sections in it
- Creates "outdated" section
- Reorganizes physics to be a walkthrough of the code (more changes to come regarding that)
- Hyperlink and referencing fixes

**Motivation and context**

This is part of an ongoing project to make the documentation as interactive and easy-to-understand as possible.

**How has this been tested?**
- [ ] Testing pipeline.
- [x] Other. 
Documentation has been built.

**Examples**

See the top of this PR for the link to the built documentation.

**Type of change**

- [ ] Bug fix. 
- [x] New feature. 
- [ ] Breaking change. 
- [ ] None of the above. 

**Checklist**

- [x] My change requires a change to the documentation.
    - [x] I have updated the documentation accordingly.
    - [x] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html).
- [x] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

File: docs/io/configuration/components/index.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+TARDIS uses the `YAML markup language <https://en.wikipedia.org/wiki/YAML>`_
+for its configuration files. There are several sections which allow different
+settings for the different aspects of the TARDIS calculation. An example
+configuration file (with a small subset of the options that can be specified) can be downloaded `here
+<https://raw.githubusercontent.com/tardis-sn/tardis/master/docs/tardis_example.yml>`_.
+
+.. warning::
+    One should note that currently floats in YAML need to be specified in a special format:
+    any pure floats need to have a +/- after the "e", e.g. 2e+5
```
Comment (andrewfullard) [inline docs/io/configuration/components/index.rst:15]: More correctly, any floats with _exponents_ need to have this format

File: docs/io/configuration/components/models/index.rst [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+******
+
+TARDIS requires a model of the homologously expanding ejecta in order to run a simulation (see :ref:`model`).
+A model will include information like the velocity shell structure, abundances, density, etc.
+TARDIS offers two ways of specifying the model: either directly in the configuration YAML file
+or separately in a model.csvy file. See `here <https://csvy.org/>`_ for an explanation of
+the CSVY file format and :ref:`here <config-file>` for a link to the csvy_model schema.
+
+.. note::
+    We highly recommend using the cleaner CSVY format.
```
Comment (andrewfullard) [inline docs/io/configuration/components/models/index.rst:14]: Not sure this is true in general, but rather for specific custom density/abundance models

File: docs/outdated/index.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
+**********************
+Outdated Documentation
+**********************
+
+The following pages are no longer relevant to the most recent release of TARDIS:
```
Comment (andrewfullard) [inline docs/outdated/index.rst:5]: Specify the release number

File: tardis/io/schemas/csvy_model.yml [hunk 4]
```diff
@@ -52,104 +53,14 @@ properties:
...
+    - $ref: 'model_definitions.yml#/definitions/density/power_law'
+    - $ref: 'model_definitions.yml#/definitions/density/uniform'
 
   abundance:
-    $ref: '#/definitions/abundance/uniform'
+    $ref: 'model_definitions.yml#/definitions/abundances/file'
 
 required:
 - tardis_model_config_version
-- model_isotope_time_0
```
Comment (andrewfullard) [inline tardis/io/schemas/csvy_model.yml:65]: Why has this all been deleted?

**Review by andrewfullard — CHANGES_REQUESTED** [review]
Generally good. Left a few comments. I hope you've tested all the new links to make sure nothing is broken.

File: tardis/io/schemas/csvy_model.yml (hunk 4 above)
Comment (smithis7) [reply tardis/io/schemas/csvy_model.yml:65]: That is related to #1594

File: docs/io/configuration/components/models/index.rst (hunk 2 above)
Comment (smithis7) [reply docs/io/configuration/components/models/index.rst:14]: What should I put there? This was copied and pasted directly from the old documentation

File: docs/io/configuration/components/index.rst (hunk 1 above)
Comment (smithis7) [reply docs/io/configuration/components/index.rst:15]: I will change that, this part was copied and pasted from the old documentation.

File: docs/outdated/index.rst (hunk 3 above)
Comment (smithis7) [reply docs/outdated/index.rst:5]: Since that is always changing, I can direct them to where to find the latest release if that works

**Review by andrewfullard — APPROVED** [review]

**Review by andrewfullard — CHANGES_REQUESTED** [review]
Please fix the test failures caused by the relocation/rename/editing of some csvy configuration files

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**